### DISCUSSION TITLE
[Desktop] Overlapping UIElements cause messaging mess up

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#17673] [Q&A]
Create two overlapping `UIElements` of your choice in `Canvas`.

In my case, this is a `Rectangle` (in background) and a `Line` (in foreground).

Do mouse click on the rectangle: 
**Expected**: `PointerPressed` handler is invoked
**Actual**: nothing happens

if you remove the `Line` from `Canvas`, the `Rectangle` reacts on mouse click as expected.

```cs
public sealed partial class MainPage : Page
{
    public MainPage() => this
            .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
            .Content(new StackPanel()
            .VerticalAlignment(VerticalAlignment.Center)
            .HorizontalAlignment(HorizontalAlignment.Center)
            .Children(
                new TextBlock()
                .Text("Hello Uno Platform!"),
                new Canvas()
                .Width(800)
                .Height(600)
                .Background(Colors.LightCyan)
                .Children(
                    new Rectangle()
                    .Width(100).Height(100)
                    .Stroke(Colors.Black)
                    .StrokeThickness(1)
                    .Fill(Colors.Pink)
                    .Canvas(100, 100)
                    .Name(ctrl => ctrl.PointerPressed += (s, e) => Debug.WriteLine("rect clicked")),
                    new Line()
                    .X1(0).Y1(0)
                    .X2(800).Y2(600)
                    .Stroke(Colors.Black)
                    .StrokeThickness(1)
                )
            ));
}
```

This issue has something to do with overlapping elements, that mess up messages.
This has nothing to do with `Canvas` layout, The issue can be observed on other layouts as well.

**TargetFrameworks:** The issue is to be observed on `Desktop`, on `Windows` all is fine.

Uno: `   "Uno.Sdk": "5.3.31"`

repro: [UnoUIElementsOverlappingApp.zip](https://github.com/user-attachments/files/16362448/UnoUIElementsOverlappingApp.zip)

---
Furthermore (not related to the main issue): if you compile and run the app on `TargetFrameworks Windows`, you may observe an exception of unknow provenience while starting
 `Exception thrown: 'System.InvalidOperationException' in WinRT.Runtime.dll`

**Comment by jeromelaban** [comment]
Let's track in https://github.com/unoplatform/uno/issues/17675
